packages/wallet-adapter-core/src/WalletCore.ts

```diff
@@ -687,9 +642,30 @@ export class WalletCore extends EventEmitter<WalletCoreEvents> {
    * @return account info
    * @throws WalletAccountError
    */
-  get account(): AccountInfo | null {
+  get account(): AccountInfoOutput | null {
     try {
-      return this._account;
+      if (!this._account) return null;
+      return {
+        address:
+          typeof this._account.address === "string"
+            ? AccountAddress.from(this._account.address)
+            : this._account.address,
+        publicKey:
+          // for backward compatibility, if the account public key is of type `string` convert it to Ed25519PublicKey
+          typeof this._account.publicKey === "string"
+            ? new Ed25519PublicKey(this._account.publicKey)
+            : // for backward compatibility, if the account public key is of type `string[]` convert each string to Ed25519PublicKey
+              Array.isArray(this._account.publicKey) &&
+                this._account.publicKey.every(
+                  (item) => typeof item === "string"
+                )
+              ? this._account.publicKey.map((key) => new Ed25519PublicKey(key))
```

Comment:
Ideally we should convert `string[]` and `minKeysRequired` to `MultiEd25519PublicKey` that extends `PublicKey`